src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBulkRecordReader.java

```diff
@@ -173,23 +199,52 @@ public void close() throws IOException {
   }
 
   @VisibleForTesting
-  void setupParser() throws IOException, AsyncApiException {
+  void setupParser() throws IOException, AsyncApiException, InterruptedException {
     if (resultIdIndex >= resultIds.length) {
       throw new IllegalArgumentException(String.format("Invalid resultIdIndex %d, should be less than %d",
-                                                       resultIdIndex, resultIds.length));
+        resultIdIndex, resultIds.length));
     }
-    InputStream queryResponseStream = bulkConnection.getQueryResultStream(jobId, batchId, resultIds[resultIdIndex]);
-
-    CSVFormat csvFormat = CSVFormat.DEFAULT
-      .withHeader()
-      .withQuoteMode(QuoteMode.ALL)
-      .withAllowMissingColumnNames(false);
-    csvParser = CSVParser.parse(queryResponseStream, StandardCharsets.UTF_8, csvFormat);
-    if (csvParser.getHeaderMap().isEmpty()) {
-      throw new IllegalStateException("Empty response was received from Salesforce, but csv header was expected.");
+    int maxRetryCount = isRetryRequired ? maxRetryAttempts : 0;
```

Comment:
when `maxRetryCount` is `0` it should not use `FailSafe` to call the salesforce query method.

This comment applies to all places in the PR.

Reply:
done the changes